simulation/traffic_simulator/include/traffic_simulator/data_type/entity_status.hpp

```diff
@@ -65,7 +65,7 @@ class CanonicalizedEntityStatus
   auto getLinearJerk() const noexcept -> double;
   auto setLinearJerk(double) -> void;
 
-  auto laneMatchingSucceed() const noexcept -> bool;
+  auto isInLanelet() const noexcept -> bool;
```

Comment:
I've added the `isInLanelet()` method to `EntityBase` with the `lanelet_id `argument because it is often used.
As result, in my feeling, it was a good thing to also change the name from `laneMatchingSucceed()` to `isInLanelet()`

Please let me explain :bow::

I based `EntityBase::isInLanelet(...)` on the `pose::isInLanelet(...)` function (from utils).
Method `pose::isInLanelet(...)` lets us check if `canonicalized_lanelet_pose` is on the lanelet with the passed `lanelet_id`:

https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/include/traffic_simulator/utils/pose.hpp#L95-L97

On the other hand, `EntityBase::isInLanelet(...)` lets us check if the `Entity` (its current position)  is on the lanelet with the passed `lanelet_id` (here is the possibility of not passing the `tolerance` value, since we can use the default value of matching):

https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp#L117-L118

https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/src/entity/entity_base.cpp#L110-L119

An additional overload of `EntityBase::isInLanelet` method can be checking if the `Entity` is on any lanelet (with any `lanelet_id`) - then we do not pass this argument as well. In this way we get a method that gives the same result as `laneMatchingSucceed()`. Because if the `Entity` is on any lanelet, then its attribute `canonicalized_lanelet_pose_` always will have a value.
So `EntityBase::isInLanelet()` without the arguments passed is equivalent to `EntityBase::laneMatchingSucceed()`


https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp#L115-L115

https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/src/data_type/entity_status.cpp#L103-L106

Reply:
I understand the background. You may resolve this conversation after reading this message.